server/akka-http-server/src/main/scala/tapir/server/akkahttp/AkkaHttpInputMatcher.scala

```diff
@@ -44,41 +33,44 @@ private[akkahttp] object AkkaHttpInputMatcher {
           case Uri.Path.Segment(s, pathTail) =>
             codec.decode(s) match {
               case DecodeResult.Value(v) =>
-                doMatch(inputsTail, ctx.withUnmatchedPath(pathTail), canRemoveSlash = true, body).map {
-                  _.prependValue(v)
-                }
+                Option(List[Any](v)) |+| doMatch(inputsTail, ctx.withUnmatchedPath(pathTail), canRemoveSlash = true, body)
```

Comment:
I also see that we have lack in our tests since this(right) is something else than the old version(left) and all tests pass.

Reply:
definitely test coverage could be better :)